src/machine/machine_stm32.go

```diff
@@ -26,14 +21,6 @@ type AltFunc uint8
 
 // ---------- General pin operations ----------
 
-// Configure this pin with the given I/O settings.
-// Remember to call enableClock() in the chip-specific implementations!
-// Also call enableAltFuncClock() if using alternative output functions when
-//  configuring peripherals, e.g. UART, SPI etc.
-func (p Pin) Configure(config PinConfig) {
-	p.configure(config)
-}
-
```

Comment:
I ended up removing this and setting the stm32f103 back to what it was before, as the other stm32 families needed an extra parameter for the alternate functions (UART, SPI etc).

Reply:
Would using interfaces for the peripherals make for a cleaner implementation?